controllers/managedosversionchannel_controller.go

```diff
@@ -297,6 +297,8 @@ func (r *ManagedOSVersionChannelReconciler) handleFailedSync(ctx context.Context
 func (r *ManagedOSVersionChannelReconciler) createManagedOSVersions(ctx context.Context, ch *elementalv1.ManagedOSVersionChannel, data []byte) error {
 	logger := ctrl.LoggerFrom(ctx)
 
+	syncTime := time.Now().UTC().String()
```

Comment:
Shouldn't we make sure we use the same value as in line 269 `ch.Status.LastSyncedTime = &now`? Would be nice having those two values being exactly the same.

I'd pass a new `syncTime *metav1.Time` argument to `createManagedOSVersions` and create the `now := metav1.Now()` variable just before calling `createManagedOSVersions`. This way would have in perfect sync the timestamps of channel status and the generated annotations in OSVersions.

Reply:
Nice catch, thank you. 
I also used the same format:

```
> kubectl -n fleet-default get managedosversionchannels elemental-channel -o yaml | yq '.status.lastSyncedTime'
2024-06-12T12:24:59Z

> kubectl -n fleet-default get managedosversion v1.1.5 -o yaml | yq '.metadata.annotations'
elemental.cattle.io/channel-last-sync: "2024-06-12T12:24:59Z"
```